chartmogul/resource.py

```diff
@@ -198,26 +199,31 @@ def _validate_arguments(cls, method, kwargs):
             )
 
     @classmethod
-    def _method(cls, method, http_verb, path=None):
+    def _method(cls, method, http_verb, path=None, customClass=None):
         @classmethod
         def fc(cls, config, **kwargs):
-            if config is None or not isinstance(config, Config):
+            klass = None
+            if customClass is None:
+                klass = cls
+            elif not issubclass(customClass, Resource):
                 raise ConfigurationError(
-                    "First argument should be" " instance of chartmogul.Config class!"
+                    "customClass should be an extension of chartmogul.Resource class!"
                 )
+            else:
+                klass = customClass
 
             pathTemp = path  # due to Python closure
             if pathTemp is None:
-                pathTemp = cls._path
+                pathTemp = klass._path
 
-            cls._validate_arguments(method, kwargs)
+            klass._validate_arguments(method, kwargs)
 
             pathTemp = Resource._expandPath(pathTemp, kwargs)
             # UUID is always path parameter only.
             if "uuid" in kwargs:
                 del kwargs["uuid"]
 
-            return cls._request(config, method, http_verb, pathTemp, **kwargs)
+            return klass._request(config, method, http_verb, pathTemp, **kwargs)
```

Comment:
I modified `._method` to allow taking a `customClass` as an argument because doing this alone will give me an error:

```py
Customer.contacts = Contact._method('all', 'get', '/customers/{uuid}/contacts')
```
The error is the response is being serialized by the `Customer` class instead. If there's a better way to do this, let me know.